coreos-base/coreos-metadata/coreos-metadata-9999.ebuild

```diff
@@ -1,36 +1,228 @@
-# Copyright (c) 2015 CoreOS, Inc.. All rights reserved.
+# Copyright (c) 2017 CoreOS, Inc.. All rights reserved.
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=6
+
 CROS_WORKON_PROJECT="coreos/coreos-metadata"
 CROS_WORKON_LOCALNAME="coreos-metadata"
 CROS_WORKON_REPO="git://github.com"
-COREOS_GO_PACKAGE="github.com/coreos/coreos-metadata"
-inherit coreos-go cros-workon systemd
 
-if [[ "${PV}" == 9999 ]]; then
+UPDATE_SSH_KEYS_VERSION="0.1.0"
+
+if [[ ${PV} == 9999 ]]; then
 	KEYWORDS="~amd64 ~arm64"
 else
-	CROS_WORKON_COMMIT="8f2578dc5231e27f05e8d105d39c03c057946862" # v0.14.0
+	CROS_WORKON_COMMIT="e28473eb9def6a9b3862a4d28549fc92be45fe59" # v1.0.0
 	KEYWORDS="amd64 arm64"
 fi
 
-DESCRIPTION="A simple cloud-provider metadata agent"
-HOMEPAGE="https://github.com/coreos/coreos-metadata"
-SRC_URI=""
+inherit coreos-cargo cros-workon systemd
 
+DESCRIPTION="A tool for collecting instance metadata from various providers"
+HOMEPAGE="https://github.com/coreos/coreos-metadata"
 LICENSE="Apache-2.0"
 SLOT="0"
-IUSE=""
 
-src_compile() {
-	export GO15VENDOREXPERIMENT="1"
-	GO_LDFLAGS="-X main.version=$(git describe --dirty)" || die
-	go_build "${COREOS_GO_PACKAGE}/internal"
+src_unpack() {
+	cros-workon_src_unpack "$@"
+	coreos-cargo_src_unpack "$@"
+	unpack update-ssh-keys-${UPDATE_SSH_KEYS_VERSION}.tar.gz
+}
+
+src_prepare() {
+	default
+
+	# check to make sure we are using the right version of update-ssh-keys
+	# our Cargo.toml should be using this version string as a tag specifier
+	grep "${UPDATE_SSH_KEYS_VERSION}" Cargo.toml || die "couldn't find update-ssh-keys version ${UPDATE_SSH_KEYS_VERSION} in Cargo.toml"
```

Comment:
The command should probably be something like `grep -q "^update-ssh-keys.*tag.*${UPDATE_SSH_KEYS_VERSION}"` so it matches the line of the expected package.

Reply:
done.